cli/runner.go

```diff
@@ -200,10 +199,18 @@ func runProcessByName(ctx context.Context, name string, settings *evergreen.Sett
 }
 
 func runnerBackgroundWorker(ctx context.Context, r processRunner, s *evergreen.Settings, dur time.Duration, wg *sync.WaitGroup) {
+	var err error
+
 	timer := time.NewTimer(0)
 	defer wg.Done()
 	defer timer.Stop()
-	defer recovery.LogStackTraceAndContinue("background runner process for", r.Name())
+	defer func() {
+		err = recovery.HandlePanicWithError(recover(), err, "worker process encountered error")
+		if err != nil {
+			wg.Add(1)
+			go runnerBackgroundWorker(ctx, r, s, dur, wg)
```

Comment:
Since this is now in a defer I don't think canceling the context will stop the goroutine will it?

Reply:
It only re-spawns if there was a panic, the context cancellation case should be safe.